### ISSUE TITLE
Dimension slider wrong round rule.

### ISSUE THREAD

**Issue by Czaki** [napari/napari#1369]
## 🐛 Bug

Napari wrong rounds on dimension slider. When image layer has scale above 1 on dimesion with toggle slider.  

## To Reproduce

1. Run napari with this code:
```python
import napari
import numpy as np

with napari.gui_qt():
    scale = (1, 209.9999999999999, 77.52248561765859, 77.52248561765859)
    viewer = napari.Viewer()
    image = np.random.random((1, 2, 100, 100))
    viewer.add_image(image, scale=(1, 100, 1, 1))
```
2. Click "New points layer"
3. Move dimension slider right (notice thar image change on 1/4 of slider, not 1/2). 

```
WARNING: Traceback (most recent call last):
  File "/home/czaki/Projekty/napari/napari/_qt/qt_dims_slider.py", line 135, in <lambda>
    lambda value: self.dims.set_point(self.axis, value)
  File "/home/czaki/Projekty/napari/napari/components/dims.py", line 359, in set_point
    self.events.axis(axis=axis, value=value)
  File "/home/czaki/Projekty/napari/napari/utils/event.py", line 508, in __call__
    self._invoke_callback(cb, event)
  File "/home/czaki/Projekty/napari/napari/utils/event.py", line 529, in _invoke_callback
    cb_event=(cb, event),
  File "/home/czaki/Projekty/napari/napari/utils/event.py", line 523, in _invoke_callback
    cb(event)
  File "/home/czaki/Projekty/napari/napari/components/viewer_model.py", line 341, in _update_layers
    layer.dims.set_point(axis, point)
  File "/home/czaki/Projekty/napari/napari/components/dims.py", line 359, in set_point
    self.events.axis(axis=axis, value=value)
  File "/home/czaki/Projekty/napari/napari/utils/event.py", line 508, in __call__
    self._invoke_callback(cb, event)
  File "/home/czaki/Projekty/napari/napari/utils/event.py", line 529, in _invoke_callback
    cb_event=(cb, event),
  File "/home/czaki/Projekty/napari/napari/utils/event.py", line 523, in _invoke_callback
    cb(event)
  File "/home/czaki/Projekty/napari/napari/layers/base/base.py", line 633, in refresh
    self.set_view_slice()
  File "/home/czaki/Projekty/napari/napari/layers/base/base.py", line 586, in set_view_slice
    self._set_view_slice()
  File "/home/czaki/Projekty/napari/napari/layers/image/image.py", line 545, in _set_view_slice
    image = np.asarray(self.data[self.dims.indices]).transpose(order)
IndexError: index 1 is out of bounds for axis 1 with size 1
[1]    23295 abort (core dumped)  python bug_reproduce.py
```

## Expected behavior
No crash

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

napari: 0.3.5.dev14+g0adb11c5
Platform: Linux-5.3.0-59-generic-x86_64-with-debian-buster-sid
Python: 3.7.2 (default, Aug 1 2019, 23:29:28) [GCC 8.3.0]
Qt: 5.15.0
PyQt5: 5.15.0
NumPy: 1.18.2
SciPy: 1.4.1
Dask: 2.12.0
VisPy: 0.6.4

GL version: 4.6.0 NVIDIA 390.132
MAX_TEXTURE_SIZE: 32768

Plugins:
- napari-plugin-engine: 0.1.5
- napari-plugins: 0.1.0
- svg: 0.1.3
 - Any other relevant information:

**Comment by sofroniewn** [comment]
Thanks for reporting @Czaki we will try and fix this as part of our efforts adding a "world" coordinate system to napari

**Comment by sofroniewn** [comment]
Closed by #1539
